pkg/blocktestservice/setup.go

```diff
@@ -0,0 +1,194 @@
+		return err
+	}
+
+	resp, err := tbs.httpClient.Do(req)
+	if err != nil {
+		tbs.logger.Errorf("Unable to fetch blocklist response: %v", err)
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusNotFound {
+		return nil
+	}
+	if resp.StatusCode != http.StatusOK {
+		err = errors.New("non 200 status")
+		tbs.logger.Errorf("Unable to fetch blocklist response: %v", err)
+		return err
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		tbs.logger.Errorf("Unable to fetch blocklist response: %v", err)
+		return err
+	}
+
+	if jsonErr := json.Unmarshal(body, &inp); jsonErr != nil {
+		tbs.logger.Errorf("Unable to fetch blocklist response: %v", jsonErr)
+		return jsonErr
+	}
+	// populate bl
+
+	blocktestLocators := make([]*blocktestLocator, 0, len(inp))
+	for i := range inp {
+		blockLocator := new(blocktestLocator)
+		blockLocator.Locator = inp[i].TestLocator
+		blockLocator.Status = inp[i].Status
+		blocktestLocators = append(blocktestLocators, blockLocator)
+	}
+	tbs.populateBlockList("api", blocktestLocators)
+	return nil
+}
+
+// GetBlockTests provides list of blocked test cases
+func (tbs *TestBlockTestService) GetBlockTests(ctx context.Context, tasConfig *core.TASConfig, repoID, branch string) error {
+
+	tbs.once.Do(func() {
+
+		blocktestLocators := make([]*blocktestLocator, 0, len(tasConfig.Blocklist))
+		for _, locator := range tasConfig.Blocklist {
+			blockLocator := new(blocktestLocator)
+			blockLocator.Locator = locator
+			blockLocator.Status = string(core.Blocklisted)
+			blocktestLocators = append(blocktestLocators, blockLocator)
+		}
+
+		tbs.populateBlockList("yml", blocktestLocators)
+
+		if err := tbs.fetchBlockListFromNeuron(ctx, repoID, branch); err != nil {
+			tbs.logger.Errorf("Unable to fetch remote blocklist: %v. Ignoring remote response", err)
+			tbs.errChan <- err
+			return
+		}
+		tbs.logger.Infof("Block tests: %+v", tbs.blockTestEntities)
+
+		// write blocklistest tests on disk
+		marshalledBlocklist, err := json.Marshal(tbs.blockTestEntities)
+		if err != nil {
+			tbs.logger.Errorf("Unable to json marshal blocklist: %+v", err)
+			tbs.errChan <- err
+			return
+		}
+
+		if err = ioutil.WriteFile(global.BlockTestFileLocation, marshalledBlocklist, 0644); err != nil {
+			tbs.logger.Errorf("Unable to write blocklist file: %+v", err)
+			tbs.errChan <- err
+			return
+		}
+		tbs.blockTestEntities = nil
+	})
+	select {
+	case err := <-tbs.errChan:
+		return err
+	default:
+		return nil
+	}
+}
+
+func (tbs *TestBlockTestService) populateBlockList(blocktestSource string, blocktestLocators []*blocktestLocator) {
+
+	i := 0
+	for _, test := range blocktestLocators {
+		//locators must end with delimiter
+		if !strings.HasSuffix(test.Locator, delimiter) {
+			test.Locator += delimiter
+		}
+		i = strings.Index(test.Locator, delimiter)
+		//TODO: handle duplicate entries and ignore its individual suites or testcases in blocklist if file is blocklisted
```

Comment:
Following code will lead to duplicate locators and we are not overriding source , we need to give precedence to api over yml